test/compilable/test17419.d

```diff
@@ -35,3 +35,19 @@ void bar()
     void nested() { }
     static assert(__traits(getLinkage, typeof(&nested)) == "D");
 }
+
+class FooD {}
+interface FooDInterface {}
+extern (C++) class FooCpp {}
+extern (C++) interface FooCppInterface {}
+
+static assert(__traits(getLinkage, FooD) == "D");
+static assert(__traits(getLinkage, FooDInterface) == "D");
+static assert(__traits(getLinkage, FooCpp) == "C++");
+static assert(__traits(getLinkage, FooCppInterface) == "C++");
```

Comment:
Please test all the linkages: https://dlang.org/spec/attribute.html#linkage

Reply:
I tested all the linkages that are supported... what do you want me to test?